middle_end/flambda2/terms/apply_expr.mli

```diff
@@ -57,7 +57,7 @@ val create :
   Debuginfo.t ->
   inlined:Inlined_attribute.t ->
   inlining_state:Inlining_state.t ->
-  probe_name:string option ->
```

Comment:
At this point we need a separate type here (in its own module), I think - the boolean is a bit mysterious otherwise.

Reply:
I added a module, but to reduce the diff I haven't made the type abstract or renamed the variables to match the new module name.

Reply:
Maybe the module could be called `Probe_name`?  Then everything looks like it will match.

Reply:
It's called just `Probe` in other IRs, and it contains not only the name, so I'd rather rename the variables if you don't mind the extra diff to review.

Reply:
done